MaterialDesignThemes.Wpf/SnackbarMessageQueueItem.cs

```diff
@@ -47,8 +48,37 @@ internal class SnackbarMessageQueueItem
         public bool IsPromoted { get; }
 
         /// <summary>
-        /// Still display this message even if it is a duplicate.
+        /// Always show this message, even if it's a duplicate
         /// </summary>
-        public bool IgnoreDuplicate { get; }
+        public bool ShowAlways { get; }
+
+        /// <summary>
+        /// Last time this message was shown
+        /// </summary>
+        public DateTime LastShownAt { get; set; }
+
+        public bool MessageExpired()
+        {
+            if (LastShownAt == null)
+                return true;
+            else
+                return LastShownAt <= DateTime.Now.Subtract(Duration);
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (!(obj is SnackbarMessageQueueItem message))
+                return false;
+
+            return Content.Equals(message.Content) && ActionContent.Equals(message.ActionContent);
```

Comment:
`Content` or `ActionContent` could be null here. I suggest using `EqualityComparer<object>.Default` to do the equality check rather `.Equals`

Reply:
Agreed. I thought about using null-conditionals, but this seems the better choice in terms of readability.